Makefile

```diff
@@ -18,6 +18,13 @@ PROJECT_DIR := $(shell dirname $(abspath $(lastword $(MAKEFILE_LIST))))
 # Produce CRDs that work back to Kubernetes 1.11 (no version conversion)
 CRD_OPTIONS ?= "crd:trivialVersions=true,preserveUnknownFields=false"
 
+GO_VERSION = $(shell go version | sed -r 's/^.*([0-9]+\.[0-9]+\.[0-9]+).*$$/\1/g')
+REQUIRED_GO_VERSION = $(shell cat go.mod | grep -E 'go [1-9]\.[0-9]+' | sed -r 's/^go ([0-9]+\.[0-9]+)$$/\1/g')
+
+ifeq (,$(findstring $(REQUIRED_GO_VERSION),$(GO_VERSION)))
```

Comment:
[+1] I remember we talked about this being one of the reasons people ahve trouble running the operator smoke-tester and build, awesome to add a check in here for it.